internal/bucket/bandwidth/reader.go

```diff
@@ -74,12 +74,16 @@ func (r *MonitoredReader) Read(buf []byte) (n int, err error) {
 		need = int(math.Min(float64(b), float64(need)))
 		tokens = need
 	}
-
+	// reduce tokens requested according to availability
```

Comment:
This would fix replication speed even under slow connections where we get frequent timeouts? On large objects?

Reply:
it will eventually hit a bottleneck if there is a mix of large/small objects - but I think this PR will reduce the timeouts since replication is no longer blocked waiting for a larger buffer size allowing slower connections to trickle through